rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -2165,6 +2033,40 @@ rmw_create_client(
   DDSDataReader * response_datareader = nullptr;
   DDSReadCondition * read_condition = nullptr;
   ConnextDynamicClientInfo * client_info = nullptr;
+  std::string request_type_name;
+  std::string response_type_name;
+  const void * untyped_request_members;
+  const void * untyped_response_members;
+
+  DDS_DomainParticipantQos participant_qos;
+  DDS_ReturnCode_t status = participant->get_qos(participant_qos);
+  if (status != DDS_RETCODE_OK) {
+    RMW_SET_ERROR_MSG("failed to get participant qos");
+    return NULL;
```

Comment:
Since this part return early it can stay above all the other variable declarations. No need to move it down.